flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/checkpoints/CheckpointConfigHandler.java

```diff
@@ -100,13 +100,15 @@ private static CheckpointConfigInfo createCheckpointConfigInfo(AccessExecutionGr
 					retentionPolicy != CheckpointRetentionPolicy.NEVER_RETAIN_AFTER_TERMINATION,
 					retentionPolicy != CheckpointRetentionPolicy.RETAIN_ON_CANCELLATION);
 
+			String stateName = checkpointCoordinatorConfiguration.getStateBackendName() == null ? "TEMPORARY_UNKNOWN" : checkpointCoordinatorConfiguration.getStateBackendName();
```

Comment:
If we query the rest before the job is running, maybe the `checkpointCoordinatorConfiguration.getStateBackendName()` will return null (please see the above comments), so transfer to `TEMPORARY_UNKNOWN` here.

Reply:
how would you query for a job before it is submitted? You'd end up with 404 in all cases.

Reply:
@zentol Thanks for the clarification. Before I submit the PR, I tried a few times to query the rest before job submitted, but the job submitting duration holds too small, all end up with a running job.
Maybe we should add a `checkNotNull` here for `stateBackendName`, what do you think.